src/server/qtquick/wsurfaceitem.cpp

```diff
@@ -744,7 +744,9 @@ void WSurfaceItem::setDelegate(QQmlComponent *newDelegate)
     if (d->delegate == newDelegate)
         return;
     d->delegate = newDelegate;
-    d->initForDelegate();
+    d->delegateIsDirty = true;
+    if (d->componentComplete)
+        d->initForDelegate();
```

Comment:
else should `initForDelegate` when componentComplete() ?

Reply:
Yes, `initSurface` will call `initForDelegate`, and `initSurface` is called in componentComplete.